src/test/server/web/test_api_server.py

```diff
@@ -0,0 +1,105 @@
+from datetime import datetime
+from freezegun import freeze_time  # type: ignore
+from ja.server.database.database import ServerDatabase
+from ja.server.web.api_server import WebRequestHandlerFactory, StatisticsWebServer
+from unittest import TestCase
+from unittest.mock import MagicMock
+
+import ja.server.web.requests as req
+import socket
+import time
+
+
+class MockBinaryIO:
+    def __init__(self) -> None:
+        self.data: str = ""
+
+    def write(self, data: bytes) -> None:
+        self.data += data.decode()
+
+
+class MockRequest(req.WebRequest):
+    def __init__(self, result: str):
+        self._result = result
+
+    def generate_report(self, database: ServerDatabase) -> str:
+        return self._result
+
+
+class WebRequestHandlerTest(TestCase):
+    def setUp(self) -> None:
+        self._handler = WebRequestHandlerFactory(database=None, mock_only=True)()
+        self._handler.send_error = MagicMock()
+        self._handler.send_response = MagicMock()
+        self._handler.send_header = MagicMock()
+        self._handler.end_headers = MagicMock()
+        self._handler.wfile = MockBinaryIO()
+
+    def test_invalid_path(self) -> None:
+        # v2 is not implemented yet :)
+        self.assertIsNone(self._handler.create_request_for_path("/v2/workmachines/workload"))
+        self.assertIsNone(self._handler.create_request_for_path("/v1/admins"))
+        self.assertIsNone(self._handler.create_request_for_path("/v1/jobs/hours/ab"))
+        self.assertIsNone(self._handler.create_request_for_path("/v1/user//jobs"))
+
+    def test_parse_correct_requests(self) -> None:
+        workload_request = self._handler.create_request_for_path("/v1/workmachines/workload")
+        self.assertIsInstance(workload_request, req.WorkMachineWorkloadRequest)
+
+        job_info_request = self._handler.create_request_for_path("/v1/jobs/abc123")
+        self.assertIsInstance(job_info_request, req.JobInformationRequest)
+        self.assertEqual(job_info_request._job_uid, "abc123")
+
+        user_jobs_request = self._handler.create_request_for_path("/v1/user/root/jobs")
+        self.assertIsInstance(user_jobs_request, req.UserJobsRequest)
+        self.assertEqual(user_jobs_request._user, "root")
+
+        with freeze_time('2020-01-01 12:00:00'):
```

Comment:
'blabla' => "blabla"

Reply:
????

Reply:
I think he means that we agreed to use double quotes instead of single quotes.